sw/device/lib/dif/dif_csrng.c

```diff
@@ -165,6 +195,30 @@ dif_csrng_result_t dif_csrng_generate(const dif_csrng_t *csrng, size_t len) {
   return write_application_command(csrng, &app_cmd);
 }
 
+dif_csrng_result_t dif_csrng_generate_end(const dif_csrng_t *csrng,
+                                          uint32_t *buf, size_t len) {
+  if (csrng == NULL || buf == NULL) {
+    return kDifCsrngBadArg;
+  }
+
+  // Wait until there is data ready.
+  while (!is_output_ready(csrng)) {
+  }
+
+  size_t rd_cnt = 0;
```

Comment:
Maybe declare this in the `for` loop as well?

Reply:
Done